datafusion/common/src/column.rs

```diff
@@ -346,6 +348,13 @@ impl From<(Option<&TableReference>, &Field)> for Column {
     }
 }
 
+/// Create a column, use qualifier and field name
+impl From<(Option<&TableReference>, &FieldRef)> for Column {
```

Comment:
This is the new API -- the rest of this PR updates the code to use it when possible